### DISCUSSION TITLE
Eks Cluster failing with Error: Failed to configure client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"

### DISCUSSION THREAD

**Discussion by vramanufidato** [gruntwork-io/knowledge-base#654] [Help]
Hi Team,

Currently facing error while deploying the eks cluster

[ecs-deploy-runner][2023-02-02T12:50:55+0000] ╷
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │ Error: Failed to configure client: exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1"
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │ 
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │   with module.eks_k8s_role_mapping.kubernetes_config_map.eks_to_k8s_role_mapping,
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │   on .terraform/modules/eks_k8s_role_mapping/modules/eks-k8s-role-mapping/main.tf line 33, in resource "kubernetes_config_map" "eks_to_k8s_role_mapping":
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │   33: resource "kubernetes_config_map" "eks_to_k8s_role_mapping" {
[ecs-deploy-runner][2023-02-02T12:50:55+0000] │ 
[ecs-deploy-runner][2023-02-02T12:50:55+0000] ╵
[ecs-deploy-runner][2023-02-02T12:50:55+0000] time=2023-02-02T12:50:55Z level=error msg=Terraform invocation failed in /tmp/tmplxdq9ono/sit/ap-south-1/sit/services/eks-cluster/.terragrunt-cache/F_mzWTkXyuI_2n_AlI9yOEmqdrg/fCw8UgbtyEKZwmAHcT_oAWkqDzs/modules/services/eks-cluster prefix=[/tmp/tmplxdq9ono/sit/ap-south-1/sit/services/eks-cluster] 
[ecs-deploy-runner][2023-02-02T12:50:55+0000]  prefix=[/tmp/tmplxdq9ono/sit] 
[ecs-deploy-runner][2023-02-02T12:50:55+0000] time=2023-02-02T12:50:55Z level=error msg=1 error occurred:
[ecs-deploy-runner][2023-02-02T12:50:55+0000] 	* exit status 1
[ecs-deploy-runner][2023-02-02T12:50:55+0000] [INFO] [infrastructure-deploy-script] 2023-02-02 12:50:55  "terragrunt plan-all" exited with code 1
[ecs-deploy-runner][2023-02-02T12:50:55+0000] Error: Could not run "terragrunt plan-all"
[ecs-deploy-runner][2023-02-02T12:50:55+0000] exit status 1
[ecs-deploy-runner][2023-02-02T12:50:55+0000] ERROR: exit status 1

As per our understanding we need awsv2 version and not awsv1 as currently the eks cluster version is 1.24 so we need the awscliv2 to be installed ?

[awsv2 not awsv1](https://github.com/aws/aws-cli/issues/6920)

This exec command requires a v2 version of the AWS CLI.

However if we install the AWS CLI with apt install awscli as many of us used to do, you get the v1 AWS CLI. v2 has completely different instructions and no apt package. And v1 works completely fine other than stuff like this, so you won't even notice you're massively out of date.

the command used is the gitlab

pip install awscli

when tried the similar command in local

root@10:~# apt install awscli 
Reading package lists... Done
root@10:~# aws --version
**aws-cli/1.18.69 Python/3.8.10 Linux/5.4.0-131-generic botocore/1.16.19**

root@10:~# /usr/local/bin/aws --version
**aws-cli/2.9.20 Python/3.9.11 Linux/5.4.0-131-generic exe/x86_64.ubuntu.20 prompt/off**

so as per our understanding the aws cli used is v1 in git lab and this version doesn't supports kubernetes 1.24 which is current version used in terraform-eks module

so to resolve this issue we need awsv2 in gitlab. will this cause any other issue in other gruntwork modules ?

---

<ins datetime="2023-02-03T05:19:40Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109861">Tracked in ticket #109861</a></p>
</ins>

**Comment by ellisonc** [comment]
I believe your problem is being caused by the removal of `client.authentication.k8s.io/v1alpha` in 1.24 rather than the version of the aws CLI.  The gruntwork module was updated to use V1 in [v0.56.2](https://github.com/gruntwork-io/terraform-aws-eks/releases/tag/v0.56.2).

**Reply by madsenwattiq** [reply]
Yes.  And since the Gruntworks modules prior to v0.56.2 completely skip v1beta1 (which seem to be prevalent in my setup on 1.21 and 1.22 clusters, you have to fork and manually edit main.tf in both eks-cluster and the eks core services modules (or upgrade beyond v0.56.2.).  Naturally, this isn't documented anywhere.
